sqlglot/generator.py

```diff
@@ -3015,6 +3020,12 @@ def renamecolumn_sql(self, expression: exp.RenameColumn) -> str:
         new_column = self.sql(expression, "to")
         return f"RENAME COLUMN{exists} {old_column} TO {new_column}"
 
+    def alterset_sql(self, expression: exp.AlterSet) -> str:
+        exprs = self.expressions(expression, flat=True)
+        exprs = exprs.lstrip(" ")
```

Comment:
This stems from T-SQLs property parsing e.g. `WithSystemVersionProperty`, `DataDeletionProperty` vs the `FILESTREAM_ON` option. The former are wrapped in `exp.Properties` which seem to be generating a leading whitespace, but the latter is wrapped in a simple `exp.Property` which prints as-is. 

I can mitigate this conflict by appending a whitespace in `exprs` only if `expression.expressions` does not have a `exp.Properties` child, if that sounds good

Reply:
Why does the `Properties` node generate leading whitespace?

Reply:
Afaict because of the `self.sep()` call in the return stmt of `Generator::root_properties()`

Reply:
Let's try to fix that, only `create_sql` uses `root_properties` (through `properties_sql`), so we could bubble that `sep()` call inside its body instead of pushing it down.

Reply:
Hmm, I think the problem is that (1) the `root_properties` always included a `self.sep()` and (2) the `with_properties` also always included a separator through `self.seg`. 

Not sure if my solution is optimal, but I decided to remove both these preemptive separators and added a check that will sep `root_properties` from `with_properties` if these aren't split by a newline